src/main/java/frc/bionic/swerve/Vision.java

```diff
@@ -0,0 +1,262 @@
+  private double cameraAngle;
+
+  // The integer value that the API uses to turn the lights on/flash/off
+  private final double lightOn = 3.0;
+  private final double lightFlash= 2.0;
+  private final double lightOff = 1.0;
+
+  // Establishes a Network Table feed from the limelight 
+  private final NetworkTable frontCamFeed;
+  private NetworkTable activeCamFeed;
+  private boolean usingFrontCam = false;
+
+  //Adds entries for shuffleboard values
+  private NetworkTableEntry sb_x_offset;
+  private NetworkTableEntry sb_y_offset;
+  private NetworkTableEntry sb_distance_to_target;
+  private NetworkTableEntry sb_target_area;
+  private NetworkTableEntry sb_target_skew;
+  private NetworkTableEntry sb_target_found;
+  private NetworkTableEntry sb_target_acquired;
+
+  private NetworkTableEntry sb_limelight_latency;
+  private NetworkTableEntry sb_front_camera;
+
+  // Whether the limelight has any valid targets (0 or 1)
+  private double tv;
+  // Horizontal Offset From Crosshair To Target (LL1: -27 degrees to 27 degrees | LL2: -29.8 to 29.8 degrees)
+  private double tx;
+  // Vertical Offset From Crosshair To Target (LL1: -20.5 degrees to 20.5 degrees | LL2: -24.85 to 24.85 degrees)
+  private double ty;
+  // Target Area (0% of image to 100% of image)
+  private double ta;
+  // Skew or rotation (-90 degrees to 0 degrees)
+  private double ts;
+  // The pipeline’s latency contribution (ms) Add at least 11ms for image capture latency.
+  private double tl; 
+
+  // Front light object
+  private double frontLight;
+  // Back light object 
+  private double backLight;
+  // The value the API uses to differentiate between what the user sees
+  private double pipeline = 0;
+
+  public Vision(double slope, double yintercept, double targetHeight, double cameraHeight, double cameraAngle) {
+    super();
+    this.slope = slope;
+    this.yintercept = yintercept;
+    this.targetHeight = targetHeight;
+    this.cameraHeight = cameraHeight;
+    this.cameraAngle = cameraAngle;
+
+    this.frontCamFeed = NetworkTableInstance.getDefault().getTable("limelight");
+    this.setFrontCamFeed();
+    this.updateTableVariables();
+  }  
+  
+  double ledState = 1;
+
+  @Override
+  public void periodic() {
+    frontCamFeed.getEntry("ledMode").setNumber(ledState);
+    frontCamFeed.getEntry("pipeline").setNumber(pipeline);
+  }
+
+  /**
+   * Updates network frontCamFeed variables.
+   */
+  private void updateTableVariables() {
+    tv = activeCamFeed.getEntry("tv").getDouble(0.0);
+    tx = activeCamFeed.getEntry("tx").getDouble(0.0);
+    ty = activeCamFeed.getEntry("ty").getDouble(0.0);
+    ta = activeCamFeed.getEntry("ta").getDouble(0.0);
+    ts = activeCamFeed.getEntry("ts").getDouble(0.0);
+    tl = activeCamFeed.getEntry("tl").getDouble(0.0);
+  }
+
+  /**
+   * Sets the led states
+  */
+  public void setLights(final double ledState){
+    this.ledState = ledState;
+  }
+
+  /**
+   * 
+   * Establishes a pipeline based on what the user sees
```

Comment:
That comment is a bit confusing. I believe this is specifying which one of the possibly many configured pipelines to use right now.